src/main/java/com/example/booksstore/mappers/BookMapper.java

```diff
@@ -27,4 +31,10 @@ default void setCategoryIds(@MappingTarget BookDto bookDto, Book book) {
                     .collect(Collectors.toSet()));
         }
     }
+
+    @Named("bookFromId")
+    default Book bookFromId(Long id, @Context BookRepository bookRepository) {
+        return bookRepository.findById(id).orElseThrow(()
```

Comment:
Its not the best idea to use repository in mapper, generally creating new book with id will be enough, then let hibernate fetch and fill other fields.
+ i dont see where you use this method, you have created "bookIdByBook" at CartItemMapper which enough